runtime/src/snapshot_utils.rs

```diff
@@ -5459,4 +5513,79 @@ mod tests {
         let next_id = bank.accounts().accounts_db.next_id.load(Ordering::Relaxed) as usize;
         assert_eq!(max_id, next_id - 1);
     }
+
+    #[test]
+    fn test_bank_from_latest_snapshot_dir() {
+        solana_logger::setup();
+        let genesis_config = GenesisConfig::default();
+        let mut bank = Arc::new(Bank::new_for_tests(&genesis_config));
+
+        let tmp_dir = tempfile::TempDir::new().unwrap();
+        let bank_snapshots_dir = tmp_dir.path();
+        let collecter_id = Pubkey::new_unique();
+        let snapshot_version = SnapshotVersion::default();
+
+        for _ in 0..3 {
+            // prepare the bank
+            bank = Arc::new(Bank::new_from_parent(&bank, &collecter_id, bank.slot() + 1));
+            bank.fill_bank_with_ticks_for_tests();
+            bank.squash();
+            bank.force_flush_accounts_cache();
+
+            // generate the bank snapshot directory for slot+1
+            let snapshot_storages = bank.get_snapshot_storages(None);
+            let slot_deltas = bank.status_cache.read().unwrap().root_slot_deltas();
+            add_bank_snapshot(
+                bank_snapshots_dir,
+                &bank,
+                &snapshot_storages,
+                snapshot_version,
+                slot_deltas,
+            )
+            .unwrap();
+            // Reserialize the snapshot dir to convert it from PRE to POST, because only the POST type can be used
+            // to construct a bank.
+            bank.accounts()
+                .accounts_db
+                .set_accounts_hash_for_tests(bank.slot(), AccountsHash(Hash::new(&[1; 32])));
+            let accounts_hash = &bank.get_accounts_hash().unwrap();
+            assert!(
+                crate::serde_snapshot::reserialize_bank_with_new_accounts_hash(
+                    bank_snapshots_dir,
+                    bank.slot(),
+                    accounts_hash,
+                    None
+                )
+            );
```

Comment:
Originally I was calling with the parameter "&AccountsHash(Hash::new(&[1; 32])))" directly, but it wail failing inside.  So I followed an example from another test calling set_accounts_hash_for_tests() ahead of it.  Let me try your approach.

Reply:
Good.  Calling with " &AccountsHash(Hash::new_unique())" succeeded.  Not sure why calling with "&AccountsHash(Hash::new(&[1; 32])" fails.  Is that because it is not unique?